clang/test/SemaCXX/unused.cpp

```diff
@@ -102,11 +102,21 @@ namespace PR33839 {
     for (auto [x] : a) { // expected-warning {{unused variable '[x]'}}
     }
   }
-  void use() { 
+  void use() {
     f<int>(); // expected-note {{instantiation of}}
     g<true>();
     g<false>();
     h<int>(); // expected-note {{instantiation of}}
   }
 }
+
+namespace maybe_unused_binding {
+
+void test() {
+  struct X { int a, b; } x;
+  auto [a [[maybe_unused]], b] = x; // expected-warning {{an attribute specifier sequence attached to a structured binding declaration is a C++2c extension}}
```

Comment:
Should we add a test to check the `[[maybe_unused]]` takes effect. 
e.g. checks whether there has `warning: unused variable '[a, b]' [-Wunused-variable]`.

I've a question about this test case. We only add `[[maybe_unused]]` to `a` , IIUC, should we also emit a warning that `unused variable 'b'`?

Reply:
This is that test.
By marking one of the member unused, we don't warn on any of the member.
This perhaps not perfect but the standard already specified that if one of the member is used, we should not warn on any of the biding

Reply:
Agree!